### PR TITLE
add tag feature to problem list

### PR DESCRIPTION
**Pull request by blackbbc**

### CODE REVIEW

**Comment by SummerWish** [comment]
tag is not a str. see vj2

File: vj4/controller/problemlist.py [hunk 1]
```diff
@@ -53,6 +53,30 @@
                             data=data)
 
 @argmethod.wrap
+async def add_tag_to(domain_id: str, lid:document.convert_doc_id, tag: str):
```
Comment (iceboy-sjtu) [inline vj4/controller/problemlist.py]: need space after "lid:"

File: vj4/controller/problemlist.py [hunk 2]
```diff
@@ -53,6 +53,30 @@
...
+  ldoc = await get(domain_id, lid)
+  if not ldoc:
+    return
+  taglst = ldoc['taglst'] if 'taglst' in ldoc else []
```
Comment (iceboy-sjtu) [inline vj4/controller/problemlist.py]: Don't create new abbr in a casual manner. Use 'tag_list'.

File: vj4/controller/problemlist.py [hunk 3]
```diff
@@ -53,6 +53,30 @@
                             data=data)
 
 @argmethod.wrap
+async def add_tag_to(domain_id: str, lid:document.convert_doc_id, tag: str):
+  ldoc = await get(domain_id, lid)
+  if not ldoc:
```
Comment (iceboy-sjtu) [inline vj4/controller/problemlist.py]: There must be something wrong if you check for existence and ignore the result.

**Comment by SummerWish** [comment]
tag should be a problem property, instead of problem list?

**Comment by SummerWish** [comment]
@twd2

**Comment by twd2** [comment]
Tags could be DOCUMENTS' property >w<!

**Pull request closed without merging**